library/include/ck/library/utility/host_tensor.hpp

```diff
@@ -254,7 +254,7 @@ struct Tensor
     Tensor(const HostTensorDescriptor& desc) : mDesc(desc), mData(mDesc.GetElementSpaceSize()) {}
 
     template <typename OutT>
-    Tensor<OutT> CopyAsType()
+    std::enable_if_t<std::is_convertible_v<T, OutT>, Tensor<OutT>> CopyAsType() const
```

Comment:
There is no alternative `CopyAsType` to fall back to even when `enable_if_t` triggers substitution failure in cases where T isn't convertible to OutT, so it fails to compile nonetheless.

It looks to me that this makes code much more verbose without adding utility here.

Reply:
The code in `CopyAsType<>()` itself made this contract:

```c++
ret.mData[i] = static_cast<OutT>(mData[i]);
```

It don't accept any type which cannot be converted to `OutT`, thus I decided to write down this assumption in the signature.

Reply:
However, the design choice, like whether to write C++ concept in template or not, is up to project culture.  
I'm okay to remote it.

Reply:
I am ok with both as the intent is clarified.